core/templates/components/state-editor/state-editor-properties-services/state-editor.service.spec.ts

```diff
@@ -145,5 +231,323 @@ describe('Editor state service', () => {
+    expect(ecs.stateHintsEditorInitialised).toBe(false);
+    ecs.updateStateHintsEditorInitialised();
+    expect(ecs.stateHintsEditorInitialised).toBe(true);
+  });
+
+  it('should update state solution editor initialised to true', () => {
+    expect(ecs.stateSolutionEditorInitialised).toBe(false);
+    ecs.updateStateSolutionEditorInitialised();
+    expect(ecs.stateSolutionEditorInitialised).toBe(true);
+  });
+
+  it('should update state editor directive initialised to true', () => {
+    expect(ecs.stateEditorDirectiveInitialised).toBe(false);
+    ecs.updateStateEditorDirectiveInitialised();
+    expect(ecs.stateEditorDirectiveInitialised).toBe(true);
+  });
+
+  it('should update update current rule input is valid', () => {
+    expect(ecs.checkCurrentRuleInputIsValid()).toBe(false);
+    ecs.updateCurrentRuleInputIsValid(true);
+    expect(ecs.checkCurrentRuleInputIsValid()).toBe(true);
+    ecs.updateCurrentRuleInputIsValid(false);
+    expect(ecs.checkCurrentRuleInputIsValid()).toBe(false);
+  });
+
+  it('should set state names and emit state names change event', () => {
+    expect(ecs.getStateNames()).toEqual([]);
+    ecs.setStateNames(['Introduction', 'State1']);
+    expect(ecs.getStateNames()).toEqual(['Introduction', 'State1']);
+    ecs.setStateNames(['Introduction', 'End']);
+    expect(ecs.getStateNames()).toEqual(['Introduction', 'End']);
+  });
+
+  it('should check event listener registration status', () => {
+    // Registration status is true only when,
+    // stateInteractionEditorInitialised, stateResponsesInitialised and
+    // stateEditorDirectiveInitialised are true.
+    expect(ecs.checkEventListenerRegistrationStatus()).toBe(false);
+
+    // Set stateInteractionEditorInitialised as true.
+    ecs.updateStateInteractionEditorInitialised();
+    expect(ecs.checkEventListenerRegistrationStatus()).toBe(false);
+
+    // Set stateResponsesInitialised as true.
+    ecs.updateStateResponsesInitialised();
+    expect(ecs.checkEventListenerRegistrationStatus()).toBe(false);
+
+    // Set stateEditorDirectiveInitialised as true.
+    ecs.updateStateEditorDirectiveInitialised();
+    expect(ecs.checkEventListenerRegistrationStatus()).toBe(true);
+  });
+
+  it('should update exploration whitelisted status', () => {
+    expect(ecs.isExplorationWhitelisted()).toBe(false);
+    ecs.updateExplorationWhitelistedStatus(true);
+    expect(ecs.isExplorationWhitelisted()).toBe(true);
+    ecs.updateExplorationWhitelistedStatus(false);
+    expect(ecs.isExplorationWhitelisted()).toBe(false);
+  });
+
+  it('should set interaction', () => {
+    expect(ecs.getInteraction()).toEqual(null);
+    ecs.setInteraction(mockInteraction);
+    expect(ecs.getInteraction()).toEqual(mockInteraction);
+  });
+
+  it('should get event emitter for change in state names', () => {
+    let stateNamesChangedEventEmitter = new EventEmitter();
+    expect(ecs.onStateNamesChanged).toEqual(
+      stateNamesChangedEventEmitter);
```

Comment:
Should we subscribe to this Observable and check that it gets called on setStateNames()?

Reply:
Done